src/pal/src/include/pal/stackstring.hpp

```diff
@@ -44,30 +58,37 @@ class StackString
             return;
         }
 
-        DeleteBuffer();
+        if( dataOnStack)
+        {
+            CopyMemory(newBuffer, m_innerBuffer, (m_count + 1) * sizeof(T));
+        }
+
         m_buffer = newBuffer;
         m_count = count;
-        m_size = count+1;
+        m_size = count_allocated + 1;
 
         return;
     }
+    
+    BOOL HasAvailableMemory(SIZE_T count)
+    {
+        return (count < m_size);
+    }
 
+    //NOTE: Always call this before modifying the underlying buffer
     void Resize(SIZE_T count)
     {
+
```

Comment:
The only time when m_buffer can be set to NULL is in the ReallocateBuffer. So I would rather change the error path that keeps the m_buffer set to NULL in there so that it resets it to point to the m_innerBuffer instead and modify Resize and ReallocateBuffer to return BOOL (success). Then this check can go away and the checks for Resize failure in the methods below would be cleaner (instead of checking the m_buffer after Resize, you would just use the return value)